src/client.rs

```diff
@@ -29,10 +34,15 @@ impl<A> ShadowTlsClient<A> {
             )
         }));
         // TLS 1.2 and TLS 1.3 is enabled.
-        let tls_config = rustls::ClientConfig::builder()
+        let mut tls_config = rustls::ClientConfig::builder()
             .with_safe_defaults()
             .with_root_certificates(root_store)
             .with_no_client_auth();
+        // Set ALPN
+        if alpn.len() != 0 {
+            tls_config.alpn_protocols = vec![alpn.as_bytes().to_vec()];
```

Comment:
I suggest put something like `TLSConfig`(with impl Default) in `ShadowTlsClient::new`, and put alpns inside it so we can extend it easily.
Also, since there can be multiple ALPN, maybe something like `Vec<Vec<u8>>`.

Reply:
It's a good idea to put all tls extension config in a struct, but i don't understand about impl Default.
After we create a struct and put config in it, `ShadowTlsClient::new` may use this struct as a parameter, so why we need default value?

Reply:
About multiple ALPN, it's easy to change it into `Vec<Vec<u8>>`, but i have no idea about how to get a vector from args. I need some help.